config/examples/redis.yml

```diff
@@ -5,6 +5,15 @@ base: &default
   sentinels: "<%= ENV['REDIS_SENTINEL_HOSTS'] %>"
   name: <%= ENV['REDIS_SENTINEL_NAME'] %>
   role: <%= ENV['REDIS_SENTINEL_ROLE'] %>
+  # == ACL Params ==
+  username: <%= ENV['REDIS_USERNAME'] %>
+  password: <%= ENV['REDIS_PASSWORD'] %>
```

Comment:
Actually it might be even better to `#to_json` them. Otherwise a password like `my nice password with " sign` will break the quoting. While `#to_json` should handle all cases.

Then `#presence` should be used on the values to check them as they will never be `nil` I assume.

Reply:
Fixed:

```yaml
# == ACL Params ==
username: <%= ENV['REDIS_USERNAME']&.to_json %>
password: <%= ENV['REDIS_PASSWORD']&.to_json %>
```

This works fine for `nil`, `>` and `"`, and I assume any other character.

Reply:
Actually `nil.to_json` results in `null` so I think the `&` is redundant.

Reply:
I tried removing the `&` and it still works. So you're right. But I'm still not sure, because:

```shell
$ bundle exec irb
irb(main):001:0> nil.to_json
Traceback (most recent call last):
...
NoMethodError (undefined method `to_json' for nil:NilClass)

irb(main):002:0> nil&.to_json
=> nil
```

But

```shell
$ bundle exec rails c
Loading development environment (Rails 6.0.6.1)
[1] pry(main)> nil.to_json
=> "null"
```

So the `&` is only redundant assuming Rails is loaded the moment the yaml is read., which I think we can assume it'll be always true. But I don't like Rails returning a string `"null"`, and sending that string to Redis, rather than a plain Ruby `nil` which we know for sure it won't cause problems on the Redis gems.

So I'd prefer to keep the `&`.

Reply:
This is a template file which should result in valid YAML which in turn would be loaded by Rails. One option would result in:
```yaml
field:
```

The other:
```yaml
field: null
```

Both result in equivalent YAML, nothing to do with rails.

```rb
🐚 pry
[1] pry(main)> require 'yaml'
=> true
[2] pry(main)> YAML.load "field: "
=> {"field"=>nil}
[3] pry(main)> YAML.load "field: null"
=> {"field"=>nil}
```

One other concern though is that either way (using `&` or not) an empty environment variable will result in the empty string `""`. While your original version would treat and empty value for the variable as `nil`. I can't immediately say whether this is a problem or not.

Reply:
Fair enough, I removed the `&`

Reply:
FYI in case it matters, to have the original behavior where unset variable is equivalent to the empty one, we could use `ENV['REDIS_PASSWORD'].presence.to_json`